src/NServiceBus.Core/Persistence/Msmq/SubscriptionStorage/MsmqSubscriptionStorage.cs

```diff
@@ -55,8 +56,9 @@ public void Init()
             }
         }
 
-        public IEnumerable<string> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes)
+        public Task<IEnumerable<string>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes)
         {
+            var subscribers = new List<string>();
             var result = new List<string>();
             var messagelist = messageTypes.ToList();
             lock (locker)
```

Comment:
@SeanFeldman it is not a dumb question. It is an excellent question. Yes any type of lock would need an async lock. I considered the MsmqStorage something legacy, plus the underlying APIs aren't async in any way. Therefore I figured throwing in an async lock is pretty much worthless. @Particular/tf-asyncawait thoughts?